config/sync/core.entity_form_display.node.article.default.yml

```diff
@@ -3,6 +3,7 @@ langcode: en
 status: true
 dependencies:
   config:
+    - field.field.node.article.field_bnf_uuid
```

Comment:
But, this is the UUID field that we'll use across sites, to keep track of which content has been imported already

Reply:
Oh right, upstream uuid.

Reply:
But I don't like that it's just a regular field. It makes a hard coupling between the module and the site configuration. Never mind the fact that we don't need it on the server.

Sadly, I don't think it's possible to hook into entities base field definition. Have you looked into what contrib modules that need to add data to entities does?

Reply:
>Never mind the fact that we don't need it on the server.

But.. we do need it on the server? It will be used when the libraries pull content from BNF?

>Sadly, I don't think it's possible to hook into entities base field definition. Have you looked into what contrib modules that need to add data to entities does?

I actually feel better about doing it this way, than adding a new dependency on another contrib module - but, if you have a suggestion for a field, I'm willing to hear it :)

Reply:
> It will be used when the libraries pull content from BNF

Server side, it's just the node UUID. What we need is an identifier that identifies a node on the server. As nodes are born with a UUID, we already have that, hell, as it only needs to be unique on the server we could use `nid`.

But if we sync UUID when we sync nodes, we don't even have to store the "original UUID" as the same content on different sites will have the same UUID (`nid` wont work for this). The only data we then need to store is the fact that this node was imported/exported, and that's just to be able to make a list of imported content.

> I actually feel better about doing it this way, than adding a new dependency on another contrib module - but, if you have a suggestion for a field, I'm willing to hear it :)

Wasn't suggesting you find a contrib module that does it for you, but that you looked into *how* they do it. `scheduler` would be a good candidate, it needs to add dates to all nodes, but they're not fields in the configuration.

Reply:
I've updated it now: https://github.com/danskernesdigitalebibliotek/dpl-cms/pull/1853/commits/3f1c7c39a69808cdd3ea05fe16d7d52b3a36ac90

Notice that I've decided to still introduce the "field" both on client and server side. This is on purpose, to avoid having to alter the code, as we both need to be able to import articles on both servers and clients.

Reply:
> This is on purpose, to avoid having to alter the code, as we both need to be able to import articles on both servers and clients.

What you've accomplished is that we now have two UUID fields on both sides, which people are bound to confuse. If we just use and sync node UUID we don't need to know *what* was synced, but *that* it was synced. Actually, for all intent and purposes, the syncing code doesn't even care about that, all it needs to know is that if two UUIDs match, it's the same content, and it should (potentially) update the receiver.

The only reason why we're interested in *that* it's synced is UI wise, to make lists of "stuff imported" and that can be handled with a simple `bnf_status` int. 0 = no BNF, 1 = incoming (imported from BNF client side, library submitted content server side) and 2 = outgoing (client side "send to BNF", not used server side). Code wise it's also much simpler to make a "list of imported content" by using `WHERE bnf_status = 1` than `WHERE bnf_uuid IS NOT NULL AND uuid != bnf_uuid`.

Reply:
I've introduced `bnf_state` now, as we've talked about.